conftest.py

```diff
@@ -20,15 +20,17 @@ def pytest_addoption(parser):
 def pytest_configure(config):
     # register an additional marker
     config.addinivalue_line(
-        "markers", "env(name): mark test to run only on named environment"
+        "markers", "env(NAME): only run test if environment NAME matches"
     )
 
 
 def pytest_runtest_setup(item):
     envnames = [mark.args[0] for mark in item.iter_markers(name="env")]
-    if envnames:
-        if item.config.getoption("-E") not in envnames:
-            pytest.skip("test requires env in %r" % envnames)
+    if (item.config.getoption("-E") is None and envnames) or (
```

Comment:
Now only tests are run either on the local target. This actually a major change in testing!!

Reply:
So this is the big unrelated change? Tests not marked are run on local, and those "local" tests are not run anymore on specific job scheduling system tests? But tests are run at least once in some environment right? 

I guess this should be OK. This might in some edge case make some bugs in CI more difficult to spot, but I don't think the situation where "local" tests pass in "no scheduling system env" and fail in "scheduling system marked env" ever occurred.

And this is because of the cryptography dependency with htcondor? It would still be better to fix this, but I imagine you already tried...

Reply:
To me the reason is not HTCondor but #521. I just pulled it into this PR because it works around the problem that there is no prebuild wheel for cryptography.  If you think that #521 is debatable, then I would remove this part and try to fix the prebuild image (e.g. by using conda/mamba to preseed the dependencies)